### ISSUE TITLE
blocked by cors policy

### ISSUE THREAD

**Issue by talwag** [typicode/jsonplaceholder#135]
Hi
i upload app to github pages and get this error :
(index):1 Access to XMLHttpRequest at 'https://jsonplaceholder.typicode.com/users' from origin 'https://talwag.github.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
xhr.js:178 GET https://jsonplaceholder.typicode.com/users net::ERR_FAILED

while working with the app local it worked ok
how can i fix it?

**Comment by marko-knoebl** [comment]
can confirm ... this happens for me as well if I run the example code from the homepage on any site (the homepage says: "Run this code in a console or from any site:")

```js
fetch('https://jsonplaceholder.typicode.com/todos/1')
  .then(response => response.json())
  .then(json => console.log(json))
```

**Comment by iamtekson** [comment]
Same error happens for me also

**Comment by JSAssassin** [comment]
I am running into the same issue 
Here's my error log: 
```
Access to XMLHttpRequest at 'https://jsonplaceholder.typicode.com/posts' from origin 'http://localhost:8080' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

**Comment by prokler** [comment]
> I am running into the same issue
> Here's my error log:
> 
> ```
> Access to XMLHttpRequest at 'https://jsonplaceholder.typicode.com/posts' from origin 'http://localhost:8080' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
> ```

Same here with 'https://jsonplaceholder.typicode.com/todos'

**Comment by talwag** [comment]
Now it is working ok although I didn't make any changes, but still the response time is bad.
that also was the case when I worked local, sometimes got cors but most of the time it worked fine.
I guess the server is very unstable.

**Comment by typicode** [comment]
Thanks for the heads up. There were some issues during the past days. It should be fixed now.

**Comment by jshinjw** [comment]
Ran into this issue today. It's still occurring with the above code:

`fetch('https://jsonplaceholder.typicode.com/todos/1')
  .then(response => response.json())
  .then(json => console.log(json))`

**Comment by sanjitkung** [comment]
I am still getting this error.

![Capture](https://user-images.githubusercontent.com/9400880/84810577-89a0c400-afd9-11ea-88d1-4956dc322562.PNG)

**Comment by frimmy** [comment]
Also get the error - might be something to do with the response being returned? [This work around has helped me](https://stackoverflow.com/questions/37121301/how-to-check-if-the-response-of-a-fetch-is-a-json-object-in-javascript) @sanjitkung - calling response.text() and then passing that value to `JSON.parse(textRes)` ... maybe the server is not returning json atm?

**Comment by back2Lobby** [comment]
I am getting same error on this simple fetch request in codepen
```
fetch('https://jsonplaceholder.typicode.com/todos/1')
.then(response => response.json())
.then(json => console.log(json))
```

**Comment by james-work-account** [comment]
@typicode I'm also getting this error calling the todos uri.

**Comment by marko-knoebl** [comment]
can confirm that it has some issues again.

It seems to be working fine if I send a request from the browser while I'm at the site, e.g. when I have https://jsonplaceholder.typicode.com opened in my browser

Also, seems to work fine if I'm at https://google.com ?

**Comment by back2Lobby** [comment]
![abc](https://user-images.githubusercontent.com/29447722/115698373-f74a1900-a37d-11eb-9ebf-96c45a1e047c.png)

**Comment by vincentherail** [comment]
Same error here
![Capture d’écran 2021-04-22 à 18 00 48](https://user-images.githubusercontent.com/62875597/115746101-56b32380-a394-11eb-9e86-fc0b6beac33d.png)

**Comment by seekingperfection** [comment]
@vincentherail Got same error..

**Comment by Med1El** [comment]
same error for me...

**Comment by masif1212** [comment]
same error for me

**Comment by doriansmiley** [comment]
Same for me

**Comment by bhanukumarnew** [comment]
same for me

**Comment by noblyblue** [comment]
https://medium.com/@dtkatz/3-ways-to-fix-the-cors-error-and-how-access-control-allow-origin-works-d97d55946d9

**Comment by doriansmiley** [comment]
This post is irrelevant. It's advocating for installing plugins and using additional services. Go to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) if you want information on CORS. This article is garbage IMO. I know I can build a proxy as a workaround. What do I need https://jsonplaceholder.typicode.com for if I build my own proxy? The answer is nothing. 

I new issue was opened two days ago [here](https://github.com/typicode/jsonplaceholder/issues/159). Anyone following this issue should bounce to the new one.

**Comment by seekingperfection** [comment]
Currently building a proxy does not solve the problem, unfortunately

**Comment by Sly-Little-Fox** [comment]
I get this error too.

**Comment by ericsonmoreira** [comment]
I get this error too.

**Comment by eyasulegesse** [comment]
try to use the proxy on you browser it works for me now

**Comment by Pyrolistical** [comment]
still a problem as of today. no cors headers
```
$ http HEAD http://jsonplaceholder.typicode.com/posts/1
HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Age: 22756
CF-Cache-Status: HIT
CF-RAY: 6e8f6aec89a48419-YVR
Cache-Control: max-age=43200
Connection: keep-alive
Content-Encoding: gzip
Content-Type: application/json; charset=utf-8
Date: Tue, 08 Mar 2022 23:40:44 GMT
Etag: W/"124-yiKdLzqO5gfBrJFrcdJ8Yq0LGnU"
Expires: -1
NEL: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
Pragma: no-cache
Report-To: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=T9TReEWsUHvF2EwX9s1kHB9s5I92sgEh%2BaEqu3YUXGKNA1IKksr3FouA6VRZHte9D%2F35c1f3UORskhVBjdFVPiK0dfIplzHi4qCwfHx8aq81Zvg5FGPU99DoeY5fXQJJwfTqWTB78I8FN9Wz25x7"}],"group":"cf-nel","max_age":604800}
Server: cloudflare
Vary: Origin, Accept-Encoding
Via: 1.1 vegur
X-Content-Type-Options: nosniff
X-Powered-By: Express
X-Ratelimit-Limit: 1000
X-Ratelimit-Remaining: 999
X-Ratelimit-Reset: 1645771009
alt-svc: h3=":443"; ma=86400, h3-29=":443"; ma=86400
```

**Comment by typicode** [comment]
I can't reproduce the issue. I tried with GET and PUT code examples from https://jsonplaceholder.typicode.com/guide/ in CodePen, JSFiddle and localhost (using the browser console).

Could someone share a failing case on CodePen or any online code editor?

**Comment by Pyrolistical** [comment]
@typicode seems to be working again. Your host must have failed and recovered

**Comment by Vimbi** [comment]
Access to fetch at 'https://jsonplaceholder-typicode-com.translate.goog/users' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

**Comment by thaikolja** [comment]
Also, CORS problem here.

**Comment by MosDevx** [comment]
Disabling the HTTPS Everywhere extension on my browser worked for me. The response was targeting http://... whilst the extension was causing the page to be https://...

**Comment by giolmartin** [comment]
Worked for me at while having the jsonplaceholders page open
